File: spark-in-action/java/src/main/java/ingestion/file/parquet/ParquetToDataframeApp.java
package ingestion.file.parquet;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;


public class ParquetToDataframeApp {
    public static void main(String[] args) {
        ParquetToDataframeApp app = new ParquetToDataframeApp();
        app.start();
    }

    /**
     * The processing code.
     */
    private void start() {
        // Creates a session on a local master
        SparkSession spark = SparkSession.builder()
                .appName("Parquet to Dataframe")
                .master("local")
                .getOrCreate();

        spark.sparkContext().setLogLevel("ERROR");

        // Reads a Parquet file, stores it in a dataframe
        Dataset<Row> df = spark.read()
                .format("parquet")
                .load("data/ch07/alltypes_plain.parquet");

        df.show(10);
        df.printSchema();
        System.out.println("The dataframe has " + df.count() + " rows.");
    }
}

File: spark-in-action/scala/src/main/scala/ingestion/file/ParquetToDataframeApp.scala
package ingestion.file

import org.apache.spark.sql.SparkSession

object ParquetToDataframeApp {

  def main(args: Array[String]): Unit = {

    val spark = SparkSession.builder
      .appName("Parquet to Dataframe")
      .master("local[*]")
      .getOrCreate

    val df = spark.read
      .format("parquet")
      .load("data/ch07/alltypes_plain.parquet")

    df.show(10)
    df.printSchema()
    println(s"The dataframe has ${df.count} rows")

    spark.stop
  }

}
